In [60]:
import nltk
nltk.download('vader_lexicon')
from nltk.sentiment import SentimentIntensityAnalyzer

import numpy as np

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\Abdullah\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [2]:
analyzer = SentimentIntensityAnalyzer()


## We start the analysis for each student

### Student 1

In [63]:
st1txt = "Things were very clear and needed a lot of thinking."

In [64]:
scores1 = analyzer.polarity_scores(st1txt)

In [65]:
print(scores1)

{'neg': 0.0, 'neu': 0.734, 'pos': 0.266, 'compound': 0.4391}


## Student 2

In [66]:
st2txt = "None at the moment."

In [67]:
scores2 = analyzer.polarity_scores(st2txt)

In [68]:
print(scores2)

{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}


## Student 3

In [69]:
st3txt = "As a whole, the chapter offers a great deal of helpful information regarding the ways in which AI can benefit our society. In addition to the references and examples used, the content of the chapter is clear and easy to understand."

In [70]:
scores3 = analyzer.polarity_scores(st3txt)

In [71]:
print(scores3)

{'neg': 0.0, 'neu': 0.688, 'pos': 0.312, 'compound': 0.9371}


## Student 4

In [72]:
st4txt = "It was helpful to look at AI from a different perspective and to understand the human-AI relationship better."

In [73]:
scores4 = analyzer.polarity_scores(st4txt)

In [74]:
print(scores4)

{'neg': 0.0, 'neu': 0.725, 'pos': 0.275, 'compound': 0.6908}


## Student 5

In [75]:
st5txt = "It was very difficult to give a blanket answer to these questions. Especially because one was not familiar with some areas."

In [76]:
scores5 = analyzer.polarity_scores(st5txt)

In [77]:
print(scores5)

{'neg': 0.128, 'neu': 0.872, 'pos': 0.0, 'compound': -0.4201}


## Student 6

In [78]:
st6txt = "The scope of the some of the questions was so broad that it almost seemed like they expected some fantasies as response. I would have preferred to have more concrete problems/arguments to be debated for or against."

In [79]:
scores6 = analyzer.polarity_scores(st6txt)

In [80]:
print(scores6)

{'neg': 0.0, 'neu': 0.94, 'pos': 0.06, 'compound': 0.3008}


## Student 7

In [81]:
st7txt = "I think this chapter is kind of difficult than the first one. Since the lecture and the exercise is a bit different."

In [82]:
scores7 = analyzer.polarity_scores(st7txt)

In [83]:
print(scores7)

{'neg': 0.128, 'neu': 0.872, 'pos': 0.0, 'compound': -0.4201}


## Student 8

In [84]:
st8txt = "Everything good."

In [85]:
scores8 = analyzer.polarity_scores(st8txt)

In [86]:
print(scores8)

{'neg': 0.0, 'neu': 0.256, 'pos': 0.744, 'compound': 0.4404}


## Student 9

In [87]:
st9txt = "I found the chapter quite difficult to answer because the questions and the scenarios are highly entwined with the basic struggles of our society."

In [88]:
scores9 = analyzer.polarity_scores(st9txt)

In [89]:
print(scores9)

{'neg': 0.201, 'neu': 0.799, 'pos': 0.0, 'compound': -0.6478}


## Student 10

In [90]:
st10txt = "The questioning and literature was quite abstract. For eg. solving poverty with AI, sharing a research work or a summary of it could be beneficial to understand as to how AI is actually applied in the field."

In [91]:
scores10 = analyzer.polarity_scores(st10txt)

In [92]:
print(scores10)

{'neg': 0.11, 'neu': 0.701, 'pos': 0.189, 'compound': 0.5267}


## Student 11

In [93]:
st11txt = "That was a bit much but very interesting and I feel like I learned something and flexed my ethics-muscle. But still I dont understand why there are deadlines at all and how these modules coincide with the presentations of our and why these exercises weren't the prerequisites for presentation, because I think this would have enhanced them."

In [94]:
scores11 = analyzer.polarity_scores(st11txt)

In [95]:
print(scores11)

{'neg': 0.0, 'neu': 0.874, 'pos': 0.126, 'compound': 0.8042}


## Student 12

In [96]:
st12txt = "The chapter gives us a summary of what we should do in terms of Ethical AI. The key information is relatable to the questions to find out the appropriate answer as well as some keywords which are very useful for online research as a clear understanding."

In [97]:
scores12 = analyzer.polarity_scores(st12txt)

In [98]:
print(scores12)

{'neg': 0.0, 'neu': 0.781, 'pos': 0.219, 'compound': 0.8805}


## Student 13

In [99]:
st13txt = "this chapter provided a more in-depth ideas about ethical issues by providing a variety of examples."

In [100]:
scores13 = analyzer.polarity_scores(st13txt)

In [101]:
print(scores13)

{'neg': 0.0, 'neu': 0.798, 'pos': 0.202, 'compound': 0.5106}


## Student 14

In [102]:
st14txt = "The Texts were clear, but the exercises weren't always too clear, not too sure what was asked of me, but it's not annoying or so.<br /> Difficulty is fine I would say."

In [103]:
scores14 = analyzer.polarity_scores(st14txt)

In [104]:
print(scores14)

{'neg': 0.136, 'neu': 0.612, 'pos': 0.252, 'compound': 0.5781}


## Student 15

In [105]:
st15txt = "This chapter draws attention to many overlooked problems that could arise with the use of AI, especially with the Healthcare system problem. Many of these seem easy at first glance but over time they become very complex and we need to really think of we would use ai to solve issues."

In [106]:
scores15 = analyzer.polarity_scores(st15txt)

In [107]:
print(scores15)

{'neg': 0.087, 'neu': 0.838, 'pos': 0.076, 'compound': 0.1027}


## Student 16

In [108]:
st16txt = "I found it hard to solve some of the questions since many answers do not only rely on AI but more on collaboration as humans and the underlying economical system."

In [109]:
scores16 = analyzer.polarity_scores(st16txt)

In [110]:
print(scores16)

{'neg': 0.041, 'neu': 0.912, 'pos': 0.047, 'compound': 0.0516}


In [111]:
# create a list of dictionaries to add
scores_list = [scores1, scores2, scores3, scores4, scores5, scores6, scores7, scores8,
               scores9, scores10, scores11, scores12, scores13, scores14, scores15, scores16]

# specify the key i need, in this case, compound
key = 'compound'

# create a list to store the values of each key that i will extract from each dict
values_list = []

# loop over the list of dictionaries and extract the values for the specific key
for d in scores_list:
    if key in d:
        values_list.append(d[key])

# calculate the average of the values:
average = np.mean(values_list)

print(average)

0.29841249999999997


# Week 2/ Chapter 2 scores are:

### The average score for week one sentiment analysis is 0.298 (1 being the most positive, 0 neutral, and -1 being negative)